### DISCUSSION TITLE
integrating go-git-providers with a library like jenkins/go-scm to increase the number of supported git providers

### DISCUSSION THREAD

**Discussion by enekofb** [fluxcd/flux2#3292] [Proposals] [answered]
Heya 

I would like to ask whether integrating go-git-providers with a library like [jenkins/go-scm](https://github.com/jenkins-x/go-scm/) would be a solution that you would like to explore in order to increase the number of supported git providers. 

An initial discovery of how this would look like could be seen in this [pr](https://github.com/enekofb/go-git-providers/pull/3/files)

Thanks!

**Comment by yiannistri** [comment, marked as answer]
Hey Eneko,

Thanks for the suggestion. We think that pursuing a solution that utilises [jenkins-x/go-scm](https://github.com/jenkins-x/go-scm/) is not optimal because:
- it is simpler to depend on SDK libraries that directly target the different git providers instead of going through an additional abstraction layer such as `go-scm` that itself targets the different git providers. After all, go-git-providers is itself an abstraction.
- SDK libraries (e.g [google/go-github](https://github.com/google/go-github), [xanzy/go-gitlab](https://github.com/xanzy/go-gitlab) etc.) are well maintained and typically provide a wide set of features that we can choose to use whereas `go-scm` provides a smaller subset of features.

We (currently) lack support of Gitea and Azure Devops but we prefer to put our efforts on adding support for those providers through SDKs (where possible) than introducing `go-scm` to fill that gap.

Hope the above makes sense.

**Comment by enekofb** [comment]
Thanks, @yiannistri for taking the time to look at my suggestion and the reply. I could totally understand your point as maintainers.
